OpenRA.Game/Actor.cs

```diff
@@ -138,6 +139,32 @@ internal Actor(World world, string name, TypeDictionary initDict)
 			SyncHashes = TraitsImplementing<ISync>().Select(sync => new SyncHash(sync)).ToArray();
 		}
 
+		internal void Created()
+		{
+			created = true;
+
+			foreach (var t in TraitsImplementing<INotifyCreated>())
+				t.Created(this);
+
+			// The initial activity should run before any activities queued by INotifyCreated.Created
+			// However, we need to know which traits are enabled (via conditions), so wait for after the calls and insert the activity as the first
+			ICreationActivity creationActivity = null;
+			foreach (var ica in TraitsImplementing<ICreationActivity>())
+			{
+				if (!ica.IsTraitEnabled())
+					continue;
+
+				if (creationActivity != null)
+					throw new InvalidOperationException("More than one enabled ICreationActivity trait: {0} and {1}".F(creationActivity.GetType().Name, ica.GetType().Name));
+
+				creationActivity = ica;
```

Comment:
Fixing #17184 requires the ability to opt out of `ICreationActivity` by returning null

```suggestion
				var activity = ica.GetCreationActivity();
				if (activity != null)
				{
					creationActivity = ica;
					activity.Queue(CurrentActivity);
					CurrentActivity = activity;
				}
```